### DISCUSSION TITLE
Consider an "unbridge" option?

### DISCUSSION THREAD

**Discussion by soulcook** [dgreif/ring#584] [Ideas]
Hi

Following our recent chats about snapshot performance, I had an interesting experience with another plugin.

I have a Ring doorbell camera, a Nest doorbell camera, and six other cameras all handled by [homebridge-camera-ffmpeg](https://github.com/Sunoo/homebridge-camera-ffmpeg). 

The camera-ffmpeg plugin (which is really good) has an option called `unbridge` on every camera. David Maher, the author of the plugin, has been recommending setting `unbridge: true` if possible to improve performance.

My cameras were all previously bridged, but I have just tried changing them all over, and the improvement is very surprising. All the cameras now provide snapshots almost instantly (it would take a variable number of seconds before) and the stream loads much more quickly too. Interestingly, the performance of the Nest and Ring cameras has improved noticeably as well -- they both fetch snapshots more quickly than they did before.

The only disadvantage of not operating in bridged mode is that the accessories no longer show up automatically in HomeKit. The user has to add them individually. But only once, of course. 

Might this be worth considering for the Ring plugin?

Paul

**Comment by dgreif** [comment]
That's a fair request and something I've considered in the past.  Originally, homebridge _only_ supported unbridged cameras which lead to user confusion when their cameras didn't automatically show up in HomeKit.  A while back (v1.0.0 I beleive), homebridge added bridged camera support and it seemed like a much better path at the time.  Only after all the plugins were converted to use bridged cameras did we realize the full impact of slowing down the bridge while waiting for snapshots and streams from the cameras.  I'm going to discuss with the core homebridge team about adding a _global_ "unbridge cameras" option that would benefit all the camera plugins instead of having each camera plugin implement it on their own.  Just checked with @Sunoo and he's on-board with that idea as well.

**Reply by soulcook** [reply]
Great. Sounds like an excellent idea. Looking forward to hearing how this discussion goes.
